### PR TITLE
MDEV-22023 Update man page NAME section to say MariaDB instead of MySQL

### PR DESCRIPTION
**Pull request by PhysicsTing**
- [x] *The Jira issue number for this PR is: MDEV-22023*

## Description
Continue the effort of a previous commit (PR#2114) which changed the man pages titles from MariaDB to MySQL, to further update the man pages.

Update the man page NAME sections to use mariadb-* instead of mysql* for MariaDB binaries that are drop-in replacements for MySQL equivalents, indicating that the commands are actually of the MariaDB version.

Before:
```
NAME
    mysql_upgrade - check tables for MariaDB upgrade
...
```

After:
```
NAME
    mariadb-upgrade - check tables for MariaDB upgrade (mysql_upgrade
    is now a symlink to mariadb-upgrade)
...
```

## How can this PR be tested?
Only man pages are changed so it is not amenable to test automation. However mtr main suite still runs and passes. Also man pages are read using man tool to verify that the outputs are as expected.

## Basing the PR against the correct MariaDB version
- [x] *This is a new feature and the PR is based against the latest MariaDB development branch*

Target 10.5 which was when that MariaDB names became default.

## Copyright
All new code of the whole pull request, including one or several files that are either new files or modified ones, are contributed under the BSD-new license. I am contributing on behalf of my employer Amazon Web Services, Inc.

### CODE REVIEW

File: man/mysqld_safe_helper.1 [hunk 1]
```diff
@@ -9,7 +9,7 @@
 .\" disable justification (adjust text to left margin only)
 .ad l
 .SH NAME
-mysqld_safe_helper \- helper script
+mariadbd-safe-helper \- helper script (mysqld_safe is now a symlink to mariadbd-safe) (mysqld_safe_helper is now a symlink to mariadbd-safe-helper)
```
Comment (iangilfillan) [inline man/mysqld_safe_helper.1:12]: This seems to be a copy-and-paste error

**Review by iangilfillan — CHANGES_REQUESTED** [review]
There are of course other places that need to be updated, but the PR is fine as an intermediate step once the pasting error is addressed.

File: man/mysqld_safe_helper.1 (hunk 1 above)
Comment (PhysicsTing) [reply man/mysqld_safe_helper.1]: Indeed. I had them fixed locally but forgot to stage it before commit&push. Thank you for pointing it out!

**Comment by PhysicsTing** [comment]
> There are of course other places that need to be updated, but the PR is fine as an intermediate step once the pasting error is addressed.
HI Ian, I addressed the pasting error. Is this PR now ready to merge or you'd like the other places to be updated together and merge together as well?

**Comment by iangilfillan** [comment]
Thanks @PhysicsTing. @grooverdan are you happy to merge?

**Comment by grooverdan** [comment]
Generally happy. The text "mysql* is now a symlink to mariadb*" becomes inaccurate in 10.6 as the symlink direction is reversed.

To aid the merging can you either do a 10.6 PR with the symlink the other way, OR change the language in this PR to say the "mysqlXXX is an alias". This I think make it accurate enough in later versions without nitpicky maintainers (:grin:) calling it incorrect.

**Comment by iangilfillan** [comment]
@grooverdan I think the text is correct as is? In 10.4, mariadb* was made a symlink to mysql*. From 10.5, the direction was reversed, with mariadb* being the names of the files, and mysql* being the symlink.

**Comment by grooverdan** [comment]
Yep, got my branches wrong. merging. Thanks @PhysicsTing

**Pull request merged**